packages/gatsby/src/utils/webpack.config.js

```diff
@@ -398,7 +398,8 @@ module.exports = async (
           // See https://github.com/gatsbyjs/gatsby/issues/11072
           styles: {
             name: `styles`,
-            test: /\.css$/,
+            // This should cover all our types of CSS.
+            test: /\.(css|scss|sass|less)$/,
```

Comment:
```suggestion
            test: /\.(css|scss|sass|less|styl)$/,
```
We also have stylus.

Some day we could traverse rules and extract `test` from ones that use `mini-css-extract`. But for quick fix this should be fine.